compiler/rustc_hir_analysis/src/collect.rs

```diff
@@ -549,6 +549,95 @@ impl<'tcx> HirTyLowerer<'tcx> for ItemCtxt<'tcx> {
         }
     }
 
+    fn lower_assoc_const(
```

Comment:
this would probably also be good to unify with `lower_assoc_ty` somehow, at the very least it would be good to not duplicate almost 100 lines of diagnostics logic :3

Reply:
Done. These two could now be on the `dyn HirTyLowerer` impl instead, with just the new `lower_assoc_shared` left as a trait method. I didn't do that though in case future impl-specific behavior was needed. Thoughts?

Reply:
Left a comment as a proper review comment, I think we can just delete both. If we ever need this/that level of flexibility it can be re-added but for now it seems unnecessary